lib/acts_as_taggable_on_mongoid/tag_list.rb

```diff
@@ -25,6 +35,10 @@ module ActsAsTaggableOnMongoid
   #   TagList.new(tag_definition, "value 1, value 2", "value 3, value 4", parser: ActsAsTaggableOnMongoid::GenericParser)
   #   # > TagList<> ["value 1", "value 2", "value 3", "value 4"]
 
+  # TODO: Address the reek: SubclassedFromCoreClass
```

Comment:
Added a note for doing this because I think we need to.

I started down the path of doing this, but it took me a while to get it somewhat working (not fully), and the risk of doing it grew too large to make it feasible to do  here/now.

Reply:
Did you add a story to rally?

Reply:
> Did you add a story to rally?

No.

I think we need to was probably a bit strong, I think we should.

This class was inherited from ActsAsTaggable originally and this is how they did it, so it probably isn't that bad (I hope) given the number of developers and users and usage of that gem.

Given that, I felt that the effort to change it doesn't give us enough business value to justify a story.

This is the kind of thing that I figured I'd do some unknown weekend in the future when I'm bored, or if it does turn out to be a problem at some point.